tests/test_xmltodict.py

```diff
@@ -221,3 +224,33 @@ def test_namespace_ignore(self):
             },
         }
         self.assertEqual(parse(xml), d)
+
+
+class StreamingGeneratorTests(unittest.TestCase):
+    def test_generator(self):
+        active_threads = threading.active_count()
+        data = '<a x="y"><b>1</b><b>2</b><b>3</b></a>'
+        result = [item for path, item in parse(data, item_depth=2)]
+        self.assertEqual(result, ['1', '2', '3'])
+        self.assertEqual(threading.active_count(), active_threads)
+
+    def test_empty_generator(self):
+        active_threads = threading.active_count()
+        data = '<a x="y"></a>'
+        result = [item for path, item in parse(data, item_depth=2)]
+        self.assertEqual(result, [])
+        self.assertEqual(threading.active_count(), active_threads)
+
+    def test_exception_handling(self):
+        active_threads = threading.active_count()
+        g = parse('', item_depth=2)
+        self.assertRaises(xml.parsers.expat.ExpatError, list, g)
+        self.assertEqual(threading.active_count(), active_threads)
+
+    def test_incomplete_iteration(self):
+        active_threads = threading.active_count()
+        data = '<a x="y"><b>1</b><b>2</b><b>3</b></a>'
+        next(parse(data, item_depth=2))
+        # TODO: unsolved issue, when parsing is not finished, but generator is
+        # no logner used, a daemonic thread is left alive.
+        self.assertEqual(threading.active_count(), active_threads + 1)
```

Comment:
@martinblech well this is very specific issue and only appears in cases when someone leaves many unfinished generators. I'm not sure what will happen if this incomplete generator gets garbage collected, maybe thread will also be removed?

I would suggest to create new ticket for this and leave it open for now. Then if someone really falls into this issue, then we could look in to it. Maybe this is not an issue at all, if it turns out, that threads are also garbage collected.

By the way, @bzamecnik [said](https://github.com/martinblech/xmltodict/issues/88#issuecomment-92136002) that he is working on a solution without threads. Maybe he succeeded with that, @bzamecnik?